octavia_f5/restclient/as3objects/service.py

```diff
@@ -111,7 +111,7 @@ def get_service(listener, cert_manager):
     }
 
     if listener.description:
-        service_args['label'] = listener.description
+        service_args['label'] = as3types.f5label(listener.description or listener.id)
```

Comment:
`listener.description` can never be `None` inside this if-branch so the label will never be set to the listener id.

Reply:
it's not about none but about ""

Reply:
If `listener.description` is "", the if-condition will never evaluate to True in the first place.
Please remove ` or listener.id`.

Reply:
Maybe we are talking past each other?
```
Python 2.7.17 (default, Dec 23 2019, 21:25:33)
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.16)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> may_be_empty = ""
>>> may_be_empty or "never empty"
'never empty'
>>> may_be_empty = "user input"
>>> may_be_empty or "never empty"
'user input'
>>>
```

Reply:
It seems we are. The point of `listener.description or listener.id` is clear.
I'm just saying that the default option `listener.id` will never be evaluated, because for that to happen, `listener.description` would need to be emptystring (or anything else that evaluates to `False`). But if it is, then the if-condition will evaluate to `False` as well and the if-body will not be entered.

If `listener.description` is not emptystring: if-body is entered, but `listener.id` is ignored.
If `listener.description` is emptystring: if-body is not entered at all.


Try it out. You can never make `foo` print 'default':
```
def foo(listener_description):
    if listener_description:
        print(listener_description or 'default')
    else:   
        print('nothing')

foo('')             # prints nothing
foo("user input")   # prints user input
```

Reply:
Ah lol, I absolutly neglected the prepending if clause! Thanks pointing out ^^